### PR TITLE
Query-frontend: add traceID to `slow query detected` log line

### PR DESCRIPTION
**Pull request by abaguas**
* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

Add traceID to `slow query detected` log line. The traceID is extracted from the `X-Thanos-Trace-Id` header, which is send in the request.
If there is no trace id the value will be empty. This behaviour is similar to the already existing log fields `grafana_dashboard_uid` and `grafana_panel_id`.

This will greatly facilitate debugging as we are often interested in analyzing the traces of the slowest queries

## Verification

Built the binary, deployed it as query frontend and checked its logs: 
```
level=info ts=2023-02-21T11:15:28.920622293Z caller=handler.go:194 org_id=anonymous msg="slow query detected" method=POST host=thanos-query-frontend-grafana.thanos.svc:9090 path=/api/v1/query time_taken=192.624733ms grafana_dashboard_uid=- grafana_panel_id=- trace_id=c33ad57d9ed443895223d5fa01499ead param_query="avg(kube_persistentvolume_capacity_bytes) by (cluster_id, persistentvolume)[1h:1m]" param_time=1676980800
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ We use *breaking :warning:* to mark changes that are not backward compatible (re
 - [#6074](https://github.com/thanos-io/thanos/pull/6074) *: Add histogram metrics `thanos_store_server_series_requested` and `thanos_store_server_chunks_requested` to all Stores.
 - [#6074](https://github.com/thanos-io/thanos/pull/6074) *: Allow configuring series and sample limits per `Series` request for all Stores.
 - [#6104](https://github.com/thanos-io/thanos/pull/6104) Objstore: Support S3 session token.
-- [#5548](https://github.com/thanos-io/thanos/pull/5548) Query: Added experimental support for load balancing across multiple Store endpoints.
```
Comment (fpetkovski) [inline CHANGELOG.md:23]: Removed by accident?

**Review by fpetkovski — APPROVED** [review]
Love it, just one small nit!

File: CHANGELOG.md (hunk 1 above)
Comment (abaguas) [reply CHANGELOG.md:23]: 🙈 Thanks, fixed it

**Review by fpetkovski — APPROVED** [review]
LGTM!

**Pull request merged**